src/backend/distributed/metadata/dependency.c

```diff
@@ -1515,6 +1631,36 @@ ExpandCitusSupportedTypes(ObjectAddressCollector *collector, ObjectAddress targe
 }
 
 
+/*
+ * ExpandForPgVanilla only expands only comosite types because other types
+ * will find their dependencies in pg_depend. The method should only be called by
+ * is_citus_depended_object udf.
+ */
+static List *
+ExpandForPgVanilla(ObjectAddressCollector *collector,
+				   ObjectAddress target)
+{
+	List *result = NIL;
+
+	if (target.classId == TypeRelationId && get_typtype(target.objectId) ==
+		TYPTYPE_COMPOSITE)
```

Comment:
Isn't ExpandCitusSupportedTypes does exactly this?

Can't we use that function? .

Reply:
It is only necessary to expand composite types for pg vanilla tests. ExpandCitusSupportedTypes expands more meta classes and causes some error logs as shown below:

```
SELECT relname, a.* FROM ONLY a, pg_class where a.tableoid = pg_class.oid;
- relname |   aa   
----------+--------
- a       | zzzz
- a       | zzzzz
- a       | zzzzzz
- a       | zzzzzz
- a       | zzzzzz
- a       | zzzzzz
-(6 rows)
-
+ERROR:  could not open relation with OID 23350
```

Reply:
It is probably a citus bug when we expand on recursion. If so, I can actually open a issue for that.